scripts/core/program.yml

```diff
@@ -10,29 +10,47 @@ type: header
 desc: "Intel $OneApi Level-Zero Runtime APIs for Program"
 ordinal: "2"
 --- #--------------------------------------------------------------------------
+type: fptr_typedef
+desc: "callback function for urProgramCreate"
+name: $x_programcreate_callback_t
+return: void
+params:
+    - type: $x_program_handle_t
+      name: hProgram
+      desc: "[in] handle of Program object created."
+    - type: void*
+      name: pParams
+      desc: "[in][out] pointer to user data to be passed to callback."
+--- #--------------------------------------------------------------------------
 type: function
 desc: "Create Program from input SPIR-V modules."
 class: $xProgram
 name: Create
 decl: static
 ordinal: "0"
 analogue:
-    - "**clCreateProgram**"
+    - "**clCreateProgramWithIL** followed by **clCompileProgram**"
 details:
     - "The application may call this function from simultaneous threads."
 params:
     - type: $x_context_handle_t
       name: hContext
       desc: "[in] handle of the context instance"
-    - type: uint32_t
-      name: count
-      desc: "[in] number of module handles in module list."
-    - type: const $x_module_handle_t*
-      name: phModules
-      desc: "[in][range(0, count)] pointer to array of modules."
+    - type: const void*
+      name: pIL
+      desc: "[in] pointer to IL string."
+    - type: size_t
+      name: length
+      desc: "[in] length of IL in bytes."
     - type: const char*
       name: pOptions
-      desc: "[in][optional] pointer to linker options null-terminated string."
+      desc: "[in] pointer to compiler options null-terminated string."
+    - type: $x_programcreate_callback_t
+      name: pfnNotify
+      desc: "[in][optional] A function pointer to a notification routine that is called when program compilation is complete."
```

Comment:
Based on the fact that no plugins currently support the callback in #9 it probably makes sense to drop this callback. Also with the addition of `ur_program_properties_t` in #275 if we wanted to add the callback back in later we could do it with a `pNext` chain.

Reply:
agreed, always nice to delete stuff if you can